File: app/src/main/java/com/jiujiu/mvvmTemplate/ui/settings/SettingFragment.java
package com.jiujiu.mvvmTemplate.ui.settings;

import android.os.Bundle;

import com.jiujiu.mvvmTemplate.R;

import androidx.preference.EditTextPreference;
import androidx.preference.ListPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceCategory;
import androidx.preference.PreferenceFragmentCompat;

public class SettingFragment extends PreferenceFragmentCompat {

    private int count = 0;

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.pref_settings, rootKey);
        setupPreference();
    }

    private void setupPreference() {
        PreferenceCategory otherCategory = findPreference("other_category");
        Preference secret_output = findPreference("secret_output");

        otherCategory.removePreference(secret_output);

        Preference secret = findPreference("secret");
        secret.setOnPreferenceClickListener(preference -> {
            if (++count >= 7) {
                otherCategory.addPreference(secret_output);
            }
            return true;
        });

        ListPreference listPreference = findPreference("sync_frequency");
        listPreference.setSummaryProvider(ListPreference.SimpleSummaryProvider.getInstance());

        EditTextPreference editTextPreference = findPreference("example_text");
        editTextPreference.setSummaryProvider(EditTextPreference.SimpleSummaryProvider.getInstance());

    }
}

File: app/src/main/java/com/jiujiu/mvvmTemplate/ui/settings/SettingFragment.kt
package com.jiujiu.mvvmTemplate.ui.settings

import android.os.Bundle
import androidx.preference.*
import com.jiujiu.mvvmTemplate.R
import com.jiujiu.mvvmTemplate.util.AppConstant.PREF_KEY_OTHER_CATEGORY
import com.jiujiu.mvvmTemplate.util.AppConstant.PREF_KEY_SECRET
import com.jiujiu.mvvmTemplate.util.AppConstant.PREF_KEY_SECRET_OUTPUT
import com.jiujiu.mvvmTemplate.util.AppConstant.PREF_KEY_SYNC_FREQUENCY

class SettingFragment : PreferenceFragmentCompat() {

    private var count = 0

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.pref_settings, rootKey)
        setupPreference()
    }

    private fun setupPreference() {
        val otherCategory = findPreference<PreferenceCategory>(PREF_KEY_OTHER_CATEGORY)
        val secretOutput = findPreference<Preference>(PREF_KEY_SECRET_OUTPUT)

        otherCategory?.removePreference(secretOutput!!)

        findPreference<Preference>(PREF_KEY_SECRET)?.apply {
            setOnPreferenceClickListener {
                if (++count >= 7) {
                    otherCategory?.addPreference(secretOutput)
                }
                true
            }
        }

        findPreference<ListPreference>(PREF_KEY_SYNC_FREQUENCY)?.summaryProvider = ListPreference.SimpleSummaryProvider.getInstance()

        findPreference<EditTextPreference>("example_text")?.summaryProvider = EditTextPreference.SimpleSummaryProvider.getInstance()
    }
}